.gitlab-ci.yml

```diff
@@ -103,13 +103,16 @@ before_script:
     - set -e
     - make -f Makefile.dune world
     - set +e
+    - tar cvfj _build.tar.bz2 _build
   variables:
     OPAM_SWITCH: edge
     OPAM_VARIANT: "+flambda"
   artifacts:
     name: "$CI_JOB_NAME"
+    when: always
```

Comment:
Not sure about this `when` , on failure some artifacts may not be found indeed.

Reply:
What about putting the tarball within the `_build/log` directory? With this hack, you make sure that the artifact path is always non-empty.